# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: modules/libpref/init/StaticPrefList.yaml
# Commit: cc843891d19e
# Full Hash: cc843891d19e329bb57eadae677211a099a9fa3e
# Author: Max Inden <mail@max-inden.de>
# Date: 2024-08-06 16:00:17
# Description:
#   Bug 1911658 - use NSPR instead of quinn-udp for QUIC I/O on Windows r=kershaw
#   
#   https://phabricator.services.mozilla.com/D217830 switched
#   `network.http.http3.use_nspr_for_io` to `false` on Firefox Nightly, thus
#   enabling the use of the newer quinn-udp instead of the older NSPR for QUIC UDP
# ==============================================================================

diff -r fbe1155fbbd7 -r cc843891d19e modules/libpref/init/StaticPrefList.yaml
--- a/modules/libpref/init/StaticPrefList.yaml	Tue Aug 06 09:27:58 2024 +0000
+++ b/modules/libpref/init/StaticPrefList.yaml	Tue Aug 06 09:41:43 2024 +0000
@@ -13564,6 +13564,11 @@
 # tracked in https://github.com/quinn-rs/quinn/issues/1947.
 #if defined(ANDROID) && !defined(HAVE_64BIT_BUILD)
   value: true
+# Always use NSPR on Windows. Windows is failing with a zero-size read. Future
+# fix and further details tracked in
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1911658.
+#elif defined(XP_WIN)
+  value: true
 #else
   value: @IS_NOT_NIGHTLY_BUILD@
 #endif
